apps/s_client.c

```diff
@@ -2322,7 +2344,27 @@ int s_client_main(int argc, char **argv)
             BIO *fbio = BIO_new(BIO_f_buffer());
 
             BIO_push(fbio, sbio);
-            BIO_printf(fbio, "CONNECT %s HTTP/1.0\r\n\r\n", connectstr);
+            BIO_printf(fbio, "CONNECT %s HTTP/1.0\r\n", connectstr);
+	    /* 
+	     * Workaround for broken proxies which would otherwise close
+	     * the connection when entering tunnel mode (eg Squid 2.6)
+	     */
+	    BIO_printf(fbio, "Proxy-Connection: Keep-Alive\r\n");
+
+            /* Support for basic (base64) proxy authentication */
+	    if (proxyuser) {
+		size_t l = strlen(proxyuser);
+		if (proxypass) l += strlen(proxypass); 
```

Comment:
don't intermix declarations and executable code.  Break the if into two lines. Use explicit NULL test.  Blank line after declarations.

Reply:
Done!